src/Linker.cs

```diff
@@ -405,6 +405,69 @@ public void DefineFunction(string module, string name, Delegate callback)
             }
         }
 
+        /// <summary>
+        /// Defines an function in the linker given an untyped callback.
+        /// </summary>
+        /// <remarks>Functions defined with this method are store-independent.</remarks>
+        /// <param name="module">The module name of the function.</param>
+        /// <param name="name">The name of the function.</param>
+        /// <param name="callback">The callback for when the function is invoked.</param>
+        /// <param name="parameters">The function parameter kinds.</param>
+        /// <param name="results">The function result kinds.</param>
+        public void DefineFunction(string module, string name, Function.UntypedCallbackDelegate callback, IReadOnlyList<ValueKind> parameters, IReadOnlyList<ValueKind> results)
+        {
+            if (module is null)
+            {
+                throw new ArgumentNullException(nameof(module));
+            }
+
+            if (name is null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (callback is null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            // Copy the lists to ensure they are not externally modified.
+            var parameterKinds = new List<ValueKind>(parameters);
+            var resultKinds = new List<ValueKind>(results);
+
+            using var funcType = Function.GetFunctionType(parameterKinds, resultKinds);
+
+            unsafe
+            {
+                Function.Native.WasmtimeFuncCallback func = (env, callerPtr, args, nargs, results, nresults) =>
+                {
+                    return Function.InvokeUntypedCallback(callback, callerPtr, args, (int)nargs, results, (int)nresults, resultKinds);
+                };
+
+                var moduleBytes = Encoding.UTF8.GetBytes(module);
```

Comment:
Do we want to choose a stackalloc limit for these strings ala #202? Defining linker functions isn't as repeated as an operation, so I don't there's a strict need.

Reply:
Agreed, I think this can then also be done for the function above with the `Delegate` parameter, so that it's consistent for all `FromCallback` overloads.

Reply:
#202 should merge any second now if you want to rebase when it does.

Reply:
I applied the changes from #202 to the remaining two overloads of `Linker.DefineFunction`, so it should be consistent now.